creatures/creatures.cpp

```diff
@@ -930,13 +981,459 @@ int Player::save(bool updateTime, LoadType saveType) {
     return(0);
 }
 
+//********************************************************************************************************
+//                                       checkResistEnchantments
+//********************************************************************************************************
+// This function can be called to check against any racial/class/other resistances to various enchantment
+// and hold spells and return true if the creature resists. Send false for output if do not want to print 
+// anything and are just checking resistances in any various logic
+
+bool Creature::checkResistEnchantments(const std::shared_ptr<Creature>& caster, const std::string spell, bool print) {
```

Comment:
Some ideas for refactoring this function for digestibility might be e.g. taking the monster type and player class checks into their own functions, or doing the `output` check once at the end, and setting the string output and broadcast params along the way (arrays if meant to print more than one thing sometimes).

Reply:
I was also thinking it was somewhat tedious. I'll break it up.

Reply:
Completed as suggested. Split into multiple functions. Easier to follow now. Also helped me find a couple bugs and even added some interesting things for output in the process.